utils/posts.ts

```diff
@@ -0,0 +1,37 @@
+import { extract } from "std/encoding/front_matter/yaml.ts";
+import { join } from "std/path/mod.ts";
+
+export interface Post {
+  slug: string;
+  title: string;
+  publishedAt: Date;
+  content: string;
+  abstract: string;
+}
+
+export async function getPosts(): Promise<Post[]> {
+  const files = Deno.readDir("./routes/blog");
```

Comment:
The blog post (`.md`) files would be best located in, say, `data/posts/` or similar rather than `routes/blog/`. The reason is that `routes/` is [specifically designed for handlers](https://fresh.deno.dev/docs/concepts/routing), not data. Are we able to move the files there?